runtime/queries/metricsview.go

```diff
@@ -292,6 +292,327 @@ func buildFilterClauseForCondition(mv *runtimev1.MetricsViewSpec, cond *runtimev
+				inHasNull = true
+				continue // Handled later using "dim IS [NOT] NULL" clause
+			}
+		}
+		inVal, subArgs, err := buildFromExpression(subExpr, allowedIdentifiers, dialect)
+		if err != nil {
+			return "", nil, err
+		}
+		args = append(args, subArgs...)
+		valClauses = append(valClauses, inVal)
+	}
+
+	// identify if immediate identifier has unnest
+	// TODO: do we need to do a deeper check?
+	unnest := false
+	ident, isIndent := cond.Exprs[0].Expression.(*runtimev1.Expression_Ident)
+	if isIndent {
+		i := allowedIdentifiers[ident.Ident]
+		unnest = i.unnest
+	}
+
+	clauses := make([]string, 0)
+
+	// If there were non-null args, add a "dim [NOT] IN (...)" clause
+	if len(valClauses) > 0 {
+		questionMarks := strings.Join(valClauses, ",")
+		var clause string
+		// Build [NOT] list_has_any("dim", ARRAY[?, ?, ...])
+		if unnest && dialect != drivers.DialectDruid {
+			clause = fmt.Sprintf("%s list_has_any(%s, ARRAY[%s])", notKeyword, leftExpr, questionMarks)
+		} else {
+			clause = fmt.Sprintf("%s %s IN (%s)", leftExpr, notKeyword, questionMarks)
+		}
+		clauses = append(clauses, clause)
+	}
+
+	if inHasNull {
+		// Add null check
+		// NOTE: DuckDB doesn't handle NULL values in an "IN" expression. They must be checked with a "dim IS [NOT] NULL" clause.
+		clauses = append(clauses, fmt.Sprintf("%s IS %s NULL", leftExpr, notKeyword))
+	}
+	var condsClause string
+	if exclude {
+		condsClause = strings.Join(clauses, " AND ")
+	} else {
+		condsClause = strings.Join(clauses, " OR ")
+	}
+	if exclude && !inHasNull && len(clauses) > 0 {
+		// When you have "dim NOT IN (a, b, ...)", then NULL values are always excluded, even if NULL is not in the list.
+		// E.g. this returns zero rows: "select * from (select 1 as a union select null as a) where a not in (1)"
+		// We need to explicitly include it.
+		condsClause += fmt.Sprintf(" OR %s IS NULL", leftExpr)
+	}
+
+	return condsClause, args, nil
+}
+
+func buildFromAndOrExpressions(cond *runtimev1.Condition, allowedIdentifiers map[string]identifier, dialect drivers.Dialect, joiner string) (string, []any, error) {
+	clauses := make([]string, 0)
+	var args []any
+	for _, expr := range cond.Exprs {
+		clause, subArgs, err := buildFromExpression(expr, allowedIdentifiers, dialect)
+		if err != nil {
+			return "", nil, err
+		}
+		args = append(args, subArgs...)
+		clauses = append(clauses, fmt.Sprintf("(%s)", clause))
+	}
+	return strings.Join(clauses, joiner), args, nil
+}
+
+func conditionExpressionOperation(oprn runtimev1.Operation) string {
+	switch oprn {
+	case runtimev1.Operation_OPERATION_EQ:
+		return "="
+	case runtimev1.Operation_OPERATION_NEQ:
+		return "!="
+	case runtimev1.Operation_OPERATION_LT:
+		return "<"
+	case runtimev1.Operation_OPERATION_LTE:
+		return "<="
+	case runtimev1.Operation_OPERATION_GT:
+		return ">"
+	case runtimev1.Operation_OPERATION_GTE:
+		return ">="
+	}
+	return "=" // TODO: handle unknown operation type
```

Comment:
Maybe panic on unknown type and add a `[(validate.rules).enum.defined_only = true]` validation to the proto requests?